app/services/configuration_repository_service.rb

```diff
@@ -14,6 +14,18 @@ class ConfigurationRepositoryService
     # @param  [ProviderCredential] provider_credential
     def initialize(provider_credential: nil)
       @client = Octokit::Client.new(access_token: provider_credential.api_token)
+      begin
```

Comment:
This is now in two places, do you think there is a way to generalize this logic so we only have it in one place?